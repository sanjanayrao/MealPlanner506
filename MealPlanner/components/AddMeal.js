import React from 'react';
import {  View, TouchableWithoutFeedback, Dimensions, TextInput, StyleSheet, ScrollView, AsyncStorage} from 'react-native';

import Button from './Button'


export default class AddMeal extends React.Component{
    constructor(props){
        super();
        this.state = {
            name: '',
            ingredients: '',
            steps: '',
            servings: '',
            user: ''
        }
    }
    setName(value){
        this.setState({name: value});
      }
      setIngredients(value){
        this.setState({ingredients: value});
      }
      setSteps(value){
        this.setState({steps: value});
      }
      setServings(value){
        this.setState({servings: value});
      }
      setUser(value){
        this.setState({user: value});
      }
    _retrieveData = async () => {
        try {
          const value = await AsyncStorage.getItem('user');
          if (value !== null) {
            // We have data!!
            this.setState({user: value})

          }
        } catch (error) {
          // Error retrieving data
        }
      };

    componentDidMount(){
        this._retrieveData()
    }

    render(){
        return(
            <View>
               
                    <TextInput
                    placeholder="Name"
                    onChangeText={value => this.setState({ name: value })}
                    />
                     <TextInput
                    placeholder="Ingredients: eg '4 cloves of garlic'"
                    onChangeText={value => this.setState({ ingredients: value })}
                    />
                     <TextInput
                    placeholder="Steps"
                    onChangeText={value => this.setState({ steps: value })}
                    />
                     <TextInput
                    placeholder="Servings"
                    onChangeText={value => this.setState({ servings: value })}
                    />
                     <Button
                        text={'Add Meal'}
                        textStyle={{color: 'white'}}
                        buttonStyle={styles.edit}
                        onPress={()=>this.updateDB() }
                    />
                
                
            </View>
        )
    }
}
const styles = StyleSheet.create({
    edit:{
        backgroundColor: '#9FC9AE', 
        padding: 10, 
        borderRadius: 10,
        marginHorizontal: 5,
        alignSelf: 'flex-end',
        alignItems: 'center',
        position: 'relative'

    },
})