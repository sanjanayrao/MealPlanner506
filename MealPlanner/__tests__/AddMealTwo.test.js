import React from 'react';
import {cleanup, fireEvent} from '@testing-library/react';
import AddMeal from '../components/AddMeal';
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure, render  } from "enzyme";







configure({ adapter: new Adapter() });



it("AddMeal Renders Same as Prior Test", () => {
    const wrapper = render(<AddMeal />);
    expect(wrapper).toMatchSnapshot();
  });
  