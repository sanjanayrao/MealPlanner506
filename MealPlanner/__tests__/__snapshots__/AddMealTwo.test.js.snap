// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMeal Renders Same as Prior Test 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "placeholder": "Name",
          "underlinecolorandroid": "transparent",
        },
        "children": Array [],
        "name": "textinput",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "placeholder": "Ingredients: eg '4 cloves of garlic'",
            "underlinecolorandroid": "transparent",
          },
          "children": Array [],
          "name": "textinput",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "placeholder": "Steps",
              "underlinecolorandroid": "transparent",
            },
            "children": Array [],
            "name": "textinput",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "placeholder": "Servings",
                "underlinecolorandroid": "transparent",
              },
              "children": Array [],
              "name": "textinput",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "focusable": "true",
                  "style": "background-color:#9FC9AE;padding:10px;border-radius:10px;margin-horizontal:5px;align-self:flex-end;align-items:center;position:relative",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "style": "color:white",
                    },
                    "children": Array [
                      Object {
                        "data": "Add Meal",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "text",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "style": undefined,
                    },
                  },
                ],
                "name": "view",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "focusable": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "focusable": undefined,
                  "style": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "placeholder": undefined,
                "underlinecolorandroid": undefined,
              },
              "x-attribsPrefix": Object {
                "placeholder": undefined,
                "underlinecolorandroid": undefined,
              },
            },
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "placeholder": undefined,
              "underlinecolorandroid": undefined,
            },
            "x-attribsPrefix": Object {
              "placeholder": undefined,
              "underlinecolorandroid": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "placeholder": undefined,
            "underlinecolorandroid": undefined,
          },
          "x-attribsPrefix": Object {
            "placeholder": undefined,
            "underlinecolorandroid": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "placeholder": undefined,
          "underlinecolorandroid": undefined,
        },
        "x-attribsPrefix": Object {
          "placeholder": undefined,
          "underlinecolorandroid": undefined,
        },
      },
      Object {
        "attribs": Object {
          "placeholder": "Ingredients: eg '4 cloves of garlic'",
          "underlinecolorandroid": "transparent",
        },
        "children": Array [],
        "name": "textinput",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "placeholder": "Steps",
            "underlinecolorandroid": "transparent",
          },
          "children": Array [],
          "name": "textinput",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "placeholder": "Servings",
              "underlinecolorandroid": "transparent",
            },
            "children": Array [],
            "name": "textinput",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "focusable": "true",
                "style": "background-color:#9FC9AE;padding:10px;border-radius:10px;margin-horizontal:5px;align-self:flex-end;align-items:center;position:relative",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "style": "color:white",
                  },
                  "children": Array [
                    Object {
                      "data": "Add Meal",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "text",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "style": undefined,
                  },
                },
              ],
              "name": "view",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "focusable": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "focusable": undefined,
                "style": undefined,
              },
            },
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "placeholder": undefined,
              "underlinecolorandroid": undefined,
            },
            "x-attribsPrefix": Object {
              "placeholder": undefined,
              "underlinecolorandroid": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "placeholder": undefined,
            "underlinecolorandroid": undefined,
          },
          "x-attribsPrefix": Object {
            "placeholder": undefined,
            "underlinecolorandroid": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "placeholder": "Name",
            "underlinecolorandroid": "transparent",
          },
          "children": Array [],
          "name": "textinput",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "placeholder": undefined,
            "underlinecolorandroid": undefined,
          },
          "x-attribsPrefix": Object {
            "placeholder": undefined,
            "underlinecolorandroid": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "placeholder": undefined,
          "underlinecolorandroid": undefined,
        },
        "x-attribsPrefix": Object {
          "placeholder": undefined,
          "underlinecolorandroid": undefined,
        },
      },
      Object {
        "attribs": Object {
          "placeholder": "Steps",
          "underlinecolorandroid": "transparent",
        },
        "children": Array [],
        "name": "textinput",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "placeholder": "Servings",
            "underlinecolorandroid": "transparent",
          },
          "children": Array [],
          "name": "textinput",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "focusable": "true",
              "style": "background-color:#9FC9AE;padding:10px;border-radius:10px;margin-horizontal:5px;align-self:flex-end;align-items:center;position:relative",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "style": "color:white",
                },
                "children": Array [
                  Object {
                    "data": "Add Meal",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "text",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "style": undefined,
                },
              },
            ],
            "name": "view",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "focusable": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "focusable": undefined,
              "style": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "placeholder": undefined,
            "underlinecolorandroid": undefined,
          },
          "x-attribsPrefix": Object {
            "placeholder": undefined,
            "underlinecolorandroid": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "placeholder": "Ingredients: eg '4 cloves of garlic'",
            "underlinecolorandroid": "transparent",
          },
          "children": Array [],
          "name": "textinput",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "placeholder": "Name",
              "underlinecolorandroid": "transparent",
            },
            "children": Array [],
            "name": "textinput",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "placeholder": undefined,
              "underlinecolorandroid": undefined,
            },
            "x-attribsPrefix": Object {
              "placeholder": undefined,
              "underlinecolorandroid": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "placeholder": undefined,
            "underlinecolorandroid": undefined,
          },
          "x-attribsPrefix": Object {
            "placeholder": undefined,
            "underlinecolorandroid": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "placeholder": undefined,
          "underlinecolorandroid": undefined,
        },
        "x-attribsPrefix": Object {
          "placeholder": undefined,
          "underlinecolorandroid": undefined,
        },
      },
      Object {
        "attribs": Object {
          "placeholder": "Servings",
          "underlinecolorandroid": "transparent",
        },
        "children": Array [],
        "name": "textinput",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "focusable": "true",
            "style": "background-color:#9FC9AE;padding:10px;border-radius:10px;margin-horizontal:5px;align-self:flex-end;align-items:center;position:relative",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "style": "color:white",
              },
              "children": Array [
                Object {
                  "data": "Add Meal",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "text",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "style": undefined,
              },
            },
          ],
          "name": "view",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "focusable": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "focusable": undefined,
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "placeholder": "Steps",
            "underlinecolorandroid": "transparent",
          },
          "children": Array [],
          "name": "textinput",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "placeholder": "Ingredients: eg '4 cloves of garlic'",
              "underlinecolorandroid": "transparent",
            },
            "children": Array [],
            "name": "textinput",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "placeholder": "Name",
                "underlinecolorandroid": "transparent",
              },
              "children": Array [],
              "name": "textinput",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "placeholder": undefined,
                "underlinecolorandroid": undefined,
              },
              "x-attribsPrefix": Object {
                "placeholder": undefined,
                "underlinecolorandroid": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "placeholder": undefined,
              "underlinecolorandroid": undefined,
            },
            "x-attribsPrefix": Object {
              "placeholder": undefined,
              "underlinecolorandroid": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "placeholder": undefined,
            "underlinecolorandroid": undefined,
          },
          "x-attribsPrefix": Object {
            "placeholder": undefined,
            "underlinecolorandroid": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "placeholder": undefined,
          "underlinecolorandroid": undefined,
        },
        "x-attribsPrefix": Object {
          "placeholder": undefined,
          "underlinecolorandroid": undefined,
        },
      },
      Object {
        "attribs": Object {
          "focusable": "true",
          "style": "background-color:#9FC9AE;padding:10px;border-radius:10px;margin-horizontal:5px;align-self:flex-end;align-items:center;position:relative",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "style": "color:white",
            },
            "children": Array [
              Object {
                "data": "Add Meal",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "text",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "style": undefined,
            },
          },
        ],
        "name": "view",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "placeholder": "Servings",
            "underlinecolorandroid": "transparent",
          },
          "children": Array [],
          "name": "textinput",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "placeholder": "Steps",
              "underlinecolorandroid": "transparent",
            },
            "children": Array [],
            "name": "textinput",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "placeholder": "Ingredients: eg '4 cloves of garlic'",
                "underlinecolorandroid": "transparent",
              },
              "children": Array [],
              "name": "textinput",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "placeholder": "Name",
                  "underlinecolorandroid": "transparent",
                },
                "children": Array [],
                "name": "textinput",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "placeholder": undefined,
                  "underlinecolorandroid": undefined,
                },
                "x-attribsPrefix": Object {
                  "placeholder": undefined,
                  "underlinecolorandroid": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "placeholder": undefined,
                "underlinecolorandroid": undefined,
              },
              "x-attribsPrefix": Object {
                "placeholder": undefined,
                "underlinecolorandroid": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "placeholder": undefined,
              "underlinecolorandroid": undefined,
            },
            "x-attribsPrefix": Object {
              "placeholder": undefined,
              "underlinecolorandroid": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "placeholder": undefined,
            "underlinecolorandroid": undefined,
          },
          "x-attribsPrefix": Object {
            "placeholder": undefined,
            "underlinecolorandroid": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "focusable": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "focusable": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "view",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
